<a href="https://colab.research.google.com/github/Benyapa-T/BSC_DPDM2025/blob/main/Midterm663020301_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1 อ่านไฟล์ทั้งหมด แสดงโครงสร้าข้อมูล และอธิบายความหมายบของแต่ละคอลัมน์

## Import Data

In [3]:
import pandas as pd
import numpy as np
import os

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


### hii_daily_rain.csv ข้อมูลปริมาณน้ำฝนรายวันย้อนหลัง

In [5]:
Daily_rain = pd.read_csv('/content/drive/MyDrive/BSC DPDM/Midterm/BSC DPDM midterm 2025/Daily rain - (Statistics 2012-2024)/hii_daily_rain.csv')
Daily_rain.head()

station_code        measure_datetime  data station_name   latitude  \
0         ABRT  2012-01-01 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
1         ABRT  2012-01-03 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
2         ABRT  2012-01-06 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
3         ABRT  2012-01-07 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
4         ABRT  2012-01-08 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   

   longitude quality_flag basin           sub_basin      tambon  \
0  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
1  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
2  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
3  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
4  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   

          amphoe  province       data_type  
0  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
1  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
2  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
3  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
4  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily

### Stations in the model_HII.csv รายชื่อสถานีที่ใช้งานได้

In [6]:
Stations_HII = pd.read_csv('/content/drive/MyDrive/BSC DPDM/Midterm/BSC DPDM midterm 2025/Input - (Operation Daily rain from API_ 2025)/Stations in the model_HII.csv')
Stations_HII.head()

Node        Lat        Long
0  ACRU  15.788054  104.642235
1  BBHN   9.536913   98.579490
2  BBUA  17.372976  103.984436
3  BCNG  17.386840  103.290870
4  BDAR  14.587073  102.495190

### Input ข้อมูลปริมาณน้ำฝนรายวันที่ต้องการนำมาประมวณผล

ตรวจสอบโครงสร้างข้อมูล

In [59]:
input_2025_07_01 = pd.read_csv('/content/drive/MyDrive/BSC DPDM/Midterm/BSC DPDM midterm 2025/Input - (Operation Daily rain from API_ 2025)/2025-07/HII_daily_rainfall_2025-07-01.csv')
input_2025_07_01.head()

rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
0  2025-07-02 00:00:00+07:00  2025-07-02T07:00:00+07:00             0.0   
1  2025-07-02 00:00:00+07:00  2025-07-02T07:00:00+07:00             0.0   
2  2025-07-02 00:00:00+07:00  2025-07-02T07:00:00+07:00            34.0   
3  2025-07-02 00:00:00+07:00  2025-07-02T07:00:00+07:00             0.2   
4  2025-07-02 00:00:00+07:00  2025-07-02T07:00:00+07:00            38.2   

   tele_station_id tele_station_oldcode measure_datetime  
0              589                 MWOG       2025-07-01  
1              564                 MSRM       2025-07-01  
2                4               CPY015       2025-07-01  
3              867                 KMSI       2025-07-01  
4               54               CPY016       2025-07-01

In [60]:
input_2025_10_01 = pd.read_csv('/content/drive/MyDrive/BSC DPDM/Midterm/BSC DPDM midterm 2025/Input - (Operation Daily rain from API_ 2025)/2025-10/HII_daily_rainfall_2025-10-01.csv')
input_2025_10_01.head()

rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
0  2025-10-02 00:00:00+07:00  2025-10-01T07:00:00+07:00             0.2   
1  2025-10-02 00:00:00+07:00  2025-10-01T07:00:00+07:00            10.8   
2  2025-10-02 00:00:00+07:00  2025-10-01T07:00:00+07:00             0.0   
3  2025-10-02 00:00:00+07:00  2025-10-01T07:00:00+07:00             2.4   
4  2025-10-02 00:00:00+07:00  2025-10-01T07:00:00+07:00             0.0   

   tele_station_id tele_station_oldcode measure_datetime  
0              812               TPI004       2025-10-01  
1              688                 TPKO       2025-10-01  
2              746               THA011       2025-10-01  
3          1198626               TRD005       2025-10-01  
4              227                 UDMS       2025-10-01

เนื่องจากข้อมูลมี Column 'measure_datetime' อยู่แล้ว สามารถทำการนำข้อมูลมาต่อกันในแนวแกน Y (Y-Data Integration) ได้เลย

In [61]:
path = '/content/drive/MyDrive/BSC DPDM/Midterm/BSC DPDM midterm 2025/Input - (Operation Daily rain from API_ 2025)'
list_month = ['2025-07','2025-08','2025-09','2025-10','2025-11','2025-12']

In [62]:
#Prompt: ต้องการดึงไฟล์ทั้งหมด ที่อยู่ใน path ออกมาจากโฟล์เดอร์แต่ละโฟล์เดอร์ list_month ทั้งหมดออกมาต่อกันเป็นไฟล์เดียว
all_df = []

for month in list_month:
    folder_path = os.path.join(path, month)

    if os.path.exists(folder_path):
        files = os.listdir(folder_path)

        for file in files:
            # เพิ่มเงื่อนไขตรวจสอบชื่อไฟล์ขึ้นต้นด้วย 'HII_daily_rainfall' และนามสกุล .csv
            if file.startswith('HII_daily_rainfall') and file.endswith('.csv'):
                file_path = os.path.join(folder_path, file)
                df = pd.read_csv(file_path)
                all_df.append(df)

        print(f"✅ โหลดไฟล์จากเดือน {month} เรียบร้อยแล้ว")

# รวมไฟล์
if all_df:
    input_df = pd.concat(all_df, ignore_index=True)
    print(f"\n✨ รวมไฟล์เสร็จสิ้น! พบข้อมูลทั้งหมด {len(input_df)} แถว")
else:
    print("\n⚠️ ไม่พบไฟล์ที่ตรงตามเงื่อนไข")

input_df

✅ โหลดไฟล์จากเดือน 2025-07 เรียบร้อยแล้ว
✅ โหลดไฟล์จากเดือน 2025-08 เรียบร้อยแล้ว
✅ โหลดไฟล์จากเดือน 2025-09 เรียบร้อยแล้ว
✅ โหลดไฟล์จากเดือน 2025-10 เรียบร้อยแล้ว
✅ โหลดไฟล์จากเดือน 2025-11 เรียบร้อยแล้ว
✅ โหลดไฟล์จากเดือน 2025-12 เรียบร้อยแล้ว

✨ รวมไฟล์เสร็จสิ้น! พบข้อมูลทั้งหมด 176886 แถว


rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
0       2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
1       2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
2       2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00            27.8   
3       2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
4       2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
...                           ...                        ...             ...   
176881  2025-12-14 00:00:00+07:00  2025-12-13T07:00:00+07:00             0.0   
176882  2025-12-14 00:00:00+07:00  2025-12-13T07:00:00+07:00             0.0   
176883  2025-12-14 00:00:00+07:00  2025-12-13T07:00:00+07:00             0.0   
176884  2025-12-14 00:00:00+07:00  2025-12-13T07:00:00+07:00             0.0   
176885  2025-12-14 00:00:00+07:00  2025-12-13T07:00:00+07:00             0.0   

        tele_station_id tele_station_oldcode measure_datetime  
0                 306.0                 BJIG       2025-07-09  
1                 466.0                 BNAN       2025-07-09  
2                 139.0                 BLKO       2025-07-09  
3                 277.0                 BNAK       2025-07-09  
4                 156.0               BPK004       2025-07-09  
...                 ...                  ...              ...  
176881            254.0                 BJAN       2025-12-13  
176882            308.0               WEI012       2025-12-13  
176883            770.0               GLF003       2025-12-13  
176884        1109588.0               ONE099       2025-12-13  
176885            286.0                 TSUD       2025-12-13  

[176886 rows x 6 columns]

### Output Y_monthly_rain_HII_operation.csv ผลลัพธ์ของข้อมูลที่ต้องการ

In [11]:
Output_HII = pd.read_csv('/content/drive/MyDrive/BSC DPDM/Midterm/BSC DPDM midterm 2025/Output - (Clean monthly rain - 2025)/Y_monthly_rain_HII_operation.csv')
Output_HII.head()

Node   rain  month  year
0  ACRU   83.0      7  2025
1  ACRU  164.8      8  2025
2  ACRU  248.6      9  2025
3  ACRU   48.4     10  2025
4  ACRU   52.2     11  2025

#2 กรองข้อมูลให้เหลือเฉพาะสถานีใน Stations in the model_HII.csv

ตรวจสอบว่า Stations in the model_HII.csv มีทั้งหมดกี่สถานี

In [63]:
Stations_HII['Node'].value_counts()

Node
YOM012    1
WNMI      1
WNKH      1
WGTG      1
WGNA      1
         ..
BDCP      1
BDAR      1
BCNG      1
BBUA      1
BBHN      1
Name: count, Length: 385, dtype: int64

แสดงให้เห็นว่ามีทั้งหมด 385 สถานี

## จำนวนสถานีก่อน drop

In [64]:
input_df.shape

(176886, 6)

In [65]:
input_df['tele_station_oldcode'].value_counts()

tele_station_oldcode
TKRA      184
TBW027    184
TKNT      184
BGBO      184
BNPK      184
         ... 
MOU256      1
MOU292      1
MOU098      1
FOP009      1
DMH018      1
Name: count, Length: 1514, dtype: int64

In [66]:
print(f"\n จำนวนข้อมูลทั้งหมด {len(input_df)} แถว")
print(f"\n จำนวนสถานีทั้งหมด {len(input_df['tele_station_oldcode'].value_counts())} สถานี")


 จำนวนข้อมูลทั้งหมด 176886 แถว

 จำนวนสถานีทั้งหมด 1514 สถานี


## จำนวนสถานีหลัง drop

In [67]:
input_df_drop = input_df[input_df['tele_station_oldcode'].isin(Stations_HII['Node'])]
input_df_drop.head()

rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
0  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
1  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
2  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00            27.8   
4  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
5  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.6   

   tele_station_id tele_station_oldcode measure_datetime  
0            306.0                 BJIG       2025-07-09  
1            466.0                 BNAN       2025-07-09  
2            139.0                 BLKO       2025-07-09  
4            156.0               BPK004       2025-07-09  
5            294.0                 BNMK       2025-07-09

In [68]:
print(f"\n จำนวนข้อมูลทั้งหมด {len(input_df_drop)} แถว")
print(f"\n จำนวนสถานีทั้งหมด {len(input_df_drop['tele_station_oldcode'].value_counts())} สถานี")


 จำนวนข้อมูลทั้งหมด 66275 แถว

 จำนวนสถานีทั้งหมด 384 สถานี


#3 ระบุข้อมูลที่สูญหายของแต่ละสถานี แสดงจำนวนและสัดส่วน

## ระบุข้อมูลที่สูญหายของแต่ละสถานี

In [69]:
input_df_not_in_stations = input_df[~input_df['tele_station_oldcode'].isin(Stations_HII['Node'])]
input_df_not_in_stations.head()

rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
3   2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
6   2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
7   2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
8   2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.2   
11  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.4   

    tele_station_id tele_station_oldcode measure_datetime  
3             277.0                 BNAK       2025-07-09  
6             254.0                 BJAN       2025-07-09  
7          575566.0               BKC002       2025-07-09  
8             720.0                 BOP1       2025-07-09  
11            259.0                 BHPK       2025-07-09

In [70]:
print(f"\n จำนวนข้อมูลทั้งหมด {len(input_df_not_in_stations)} แถว")
print(f"\n จำนวนสถานีทั้งหมด {len(input_df_not_in_stations['tele_station_oldcode'].value_counts())} สถานี")


 จำนวนข้อมูลทั้งหมด 110611 แถว

 จำนวนสถานีทั้งหมด 1130 สถานี


In [71]:
input_df_not_in_stations['tele_station_oldcode'].value_counts()

tele_station_oldcode
BNAK      184
TBW027    184
SWR029    184
TLMU      184
PAS010    184
         ... 
MOU256      1
MOU252      1
MOU098      1
FOP009      1
DMH018      1
Name: count, Length: 1130, dtype: int64

## แสดงจำนวนและสัดส่วน

In [72]:
# Prompt:ต้องการสร้าง pie chart แสดงสัดส่วนของ input_df_drop กับ input_df_not_in_stations
# โดยใช้ plotly และแสดงผลเป็นตัวเลขและเปอร์เซ็น
import plotly.express as px
import pandas as pd

# 1. เตรียมข้อมูลในรูปแบบ DataFrame (เพื่อให้ Plotly อ่านง่าย)
data = {
    'Category': ['Remaining Data', 'Dropped Data'],
    'Count': [len(input_df_drop), len(input_df_not_in_stations)]
}
df_plot = pd.DataFrame(data)

# 2. สร้าง Pie Chart
fig = px.pie(df_plot,
             values='Count',
             names='Category',
             title='Data Distribution Analysis',
             color_discrete_sequence=['#ff9999','#66b3ff','#99ff99'],
             hole=0.4) # ใส่ hole เพื่อให้เป็น Donut Chart (ถ้าไม่ใส่จะเป็น Pie ปกติ)

# 3. ตั้งค่าให้แสดงทั้งข้อความ, จำนวน และเปอร์เซ็นต์พร้อมกัน
fig.update_traces(textinfo='label+value+percent',
                  textfont_size=14,
                  marker=dict(line=dict(color='#000000', width=1)))

fig.show()

#4 คัดสถานีที่มีข้อมูลสูญหาย ≥20% ออก แสดงรายชื่อที่ถูกคัดออกและจำนวนที่เหลือ

## คัดสถานีที่มีข้อมูลสูญหาย ≥20% ออก

In [73]:
input_df_drop.isnull().sum()

rainfall_datetime            0
rainfall_datetime_calc       0
rainfall_value               1
tele_station_id           1023
tele_station_oldcode         0
measure_datetime             0
dtype: int64

In [74]:
import pandas as pd

# 1. สร้างตารางสรุปโดย Group ตาม tele_station_oldcode
# เราจะนับจำนวนแถวทั้งหมด (Total) และจำนวนค่าว่าง (Missing)
summary_df = input_df_drop.groupby('tele_station_oldcode').agg(
    total_rows=('tele_station_oldcode', 'count'),
    missing_rows=('tele_station_id', lambda x: x.isnull().sum()) # เปลี่ยน 'rainfall_value' เป็นชื่อ column ที่ต้องการเช็ค
).reset_index()

# 2. คำนวณเปอร์เซ็นต์ Missing
summary_df['missing_percentage'] = (summary_df['missing_rows'] / summary_df['total_rows']) * 100

# 3. เรียงลำดับจากสถานีที่ Missing เยอะที่สุดไปน้อยที่สุด
summary_df = summary_df.sort_values(by='missing_percentage', ascending=False)

# แสดงผล 10 อันดับแรก
print("ตารางสรุปข้อมูลแยกตามรหัสสถานี (10 อันดับแรกที่มี Missing สูงสุด):")
print(summary_df.head(10).to_string(index=False))

# 4. สรุปภาพรวม
print(f"\nจำนวนสถานีทั้งหมด: {len(summary_df)} สถานี")

ตารางสรุปข้อมูลแยกตามรหัสสถานี (10 อันดับแรกที่มี Missing สูงสุด):
tele_station_oldcode  total_rows  missing_rows  missing_percentage
                MEWG          77             3            3.896104
                CEHM          84             3            3.571429
              YOM001          99             3            3.030303
              SKE001         137             3            2.189781
                NKSG         138             3            2.173913
              TPI007         138             3            2.173913
              VLGE33         138             3            2.173913
              KON005         139             3            2.158273
              CPY010         142             3            2.112676
                BLKO         143             3            2.097902

จำนวนสถานีทั้งหมด: 384 สถานี


เนื่องจากไม่มีสถานีที่ข้อมูลสูญหาย ≥20% จึงไม่ได้ทำการคัดออก

#5 เติมข้อมูลที่หายไปด้วยค่า Medain ตาม Day of year จาก histrorical แยกตามสถานี ตรวจสอบว่า ไม่มี record หายอีก

In [80]:
input_df_drop.isnull().sum()

rainfall_datetime            0
rainfall_datetime_calc       0
rainfall_value               0
tele_station_id           1023
tele_station_oldcode         0
measure_datetime             0
dtype: int64

เนื่องจาก column 'rainfall_value' มี missing เพียง 1 ตัว ใช้ .notnull()

In [81]:
input_df_drop = input_df_drop[input_df_drop['rainfall_value'].notnull()]

In [82]:
input_df_drop[input_df_drop['rainfall_value'].isnull()]

Empty DataFrame
Columns: [rainfall_datetime, rainfall_datetime_calc, rainfall_value, tele_station_id, tele_station_oldcode, measure_datetime]
Index: []

ใช้ค่า Mode tele_station_id

In [83]:
Daily_rain.head()

station_code        measure_datetime  data station_name   latitude  \
0         ABRT  2012-01-01 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
1         ABRT  2012-01-03 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
2         ABRT  2012-01-06 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
3         ABRT  2012-01-07 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   
4         ABRT  2012-01-08 07:00:00+07   0.0   วัดเวฬุวัน  16.054785   

   longitude quality_flag basin           sub_basin      tambon  \
0  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
1  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
2  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
3  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   
4  103.66429            U    ชี  ลำน้ำชีส่วนที่ 4/2  เหนือเมือง   

          amphoe  province       data_type  
0  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
1  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
2  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
3  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily  
4  เมืองร้อยเอ็ด  ร้อยเอ็ด  rainfall_daily

#6 อภิปรายเหตุผลที่ใช้ Median แทน Mean

#7 คำนวณปริมาณน้ำฝนรายเดือน

In [85]:
Output_HII.head()

Node   rain  month  year
0  ACRU   83.0      7  2025
1  ACRU  164.8      8  2025
2  ACRU  248.6      9  2025
3  ACRU   48.4     10  2025
4  ACRU   52.2     11  2025

In [86]:
input_df_drop.head()

rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
0  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
1  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
2  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00            27.8   
4  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
5  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.6   

   tele_station_id tele_station_oldcode measure_datetime  
0            306.0                 BJIG       2025-07-09  
1            466.0                 BNAN       2025-07-09  
2            139.0                 BLKO       2025-07-09  
4            156.0               BPK004       2025-07-09  
5            294.0                 BNMK       2025-07-09

สร้าง column month year

In [87]:
# Prompt : หา เดือน ปี จาก column measure_datetime ในตาราง input_df_drop ที่มีการบันทึกข้อมูลเป็น 2025-07-09 ปี-เดือน-วัน โดยใช้ฟังก์ชั่น split

# สร้างคอลัมน์ 'month'
input_df_drop['month'] = input_df_drop['measure_datetime'].astype(str).str.split('-').str[1]

# สร้างคอลัมน์ 'year'
input_df_drop['year'] = input_df_drop['measure_datetime'].astype(str).str.split('-').str[0]

In [88]:
input_df_drop.head()

rainfall_datetime     rainfall_datetime_calc  rainfall_value  \
0  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
1  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
2  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00            27.8   
4  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.0   
5  2025-07-10 00:00:00+07:00  2025-07-10T07:00:00+07:00             0.6   

   tele_station_id tele_station_oldcode measure_datetime month  year  
0            306.0                 BJIG       2025-07-09    07  2025  
1            466.0                 BNAN       2025-07-09    07  2025  
2            139.0                 BLKO       2025-07-09    07  2025  
4            156.0               BPK004       2025-07-09    07  2025  
5            294.0                 BNMK       2025-07-09    07  2025

In [89]:
# Prompt: สร้างตารางใหม่ ชื่อ Output แล้วดึงข้อมูล tele_station_oldcode จากตาราง input_df_drop มาสร้างใหม่ชื่อว่า Node โดยข้อมูลที่ดึงมาต้องไม่ซ้ำกัน
Output = pd.DataFrame({
    'Node': input_df_drop['tele_station_oldcode'],
    'year': input_df_drop['year'],
    'month': input_df_drop['month']
})
Output.drop_duplicates()

Node  year month
0         BJIG  2025    07
1         BNAN  2025    07
2         BLKO  2025    07
4       BPK004  2025    07
5         BNMK  2025    07
...        ...   ...   ...
146591  STH011  2025    12
146596  CHR004  2025    12
146612  STU001  2025    12
146707    NKSG  2025    12
147049    BBUA  2025    12

[2212 rows x 3 columns]

In [90]:
rain_sum = (
    input_df_drop
    .groupby(['year', 'month'])['rainfall_value']
    .sum()
    .reset_index()
)

rain_sum = rain_sum.rename(columns={'rainfall_value': 'rain'})
Output = Output.merge(rain_sum, on=['year','month'], how='left')

In [91]:
Output = Output.sort_values(by='Node').reset_index(drop=True)
Output

Node  year month     rain
0        ACRU  2025    07  43621.6
1        ACRU  2025    12   6977.4
2        ACRU  2025    08  47936.6
3        ACRU  2025    11  44888.8
4        ACRU  2025    08  47936.6
...       ...   ...   ...      ...
66269  YOM012  2025    12   6977.4
66270  YOM012  2025    10  32462.6
66271  YOM012  2025    09  59587.4
66272  YOM012  2025    07  43621.6
66273  YOM012  2025    10  32462.6

[66274 rows x 4 columns]

In [93]:
path = '/content/drive/MyDrive/BSC DPDM/Midterm'
Output.to_csv(f'{path}/Output.csv', index=False)


# รวมไฟล์

* ไฟล์ colap https://colab.research.google.com/drive/10i5GEtvSSEvz1QXorugeo0TGlMEJ523D?usp=sharing
* ไฟล์ผลลัพธ์
https://drive.google.com/file/d/1eykbYmSeODpGahjp6UvqR1hjJ37fsnfV/view?usp=sharing
* ไฟล์การใช้ AI
https://docs.google.com/document/d/1-wg4Q0e-Dp_1e_1nvhZ8xBplO1pmj1EREL_GDEXzRNA/edit?usp=sharing
